### PR TITLE
cleaning up phpdoc on query builder

### PR DESCRIPTION
**Pull request by dbu**
I noticed that we have some typos in the docs. Went over them and also cleaned other codestyle issues.

### CODE REVIEW

File: lib/Doctrine/ODM/PHPCR/Query/Builder/AbstractNode.php [hunk 1]
```diff
@@ -142,9 +142,6 @@ public function addChild(AbstractNode $node)
         $cardinalityMap = $this->getCardinalityMap();
         $nodeType = $node->getNodeType();
 
-        $validChild = true;
-        $end = false;
```
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/AbstractNode.php]: unused code

File: lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php [hunk 2]
```diff
@@ -527,6 +530,9 @@ protected function doWalkConstraintComposite(AbstractNode $node, $method)
 
             $lPhpcrConstraint = $phpcrComposite;
         }
+        if (!$phpcrComposite) {
+            throw new \InvalidArgumentException('Composite must have at least one constraint');
+        }
```
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php]: this is an actual code change!
my IDE warned that $phpcrComposite might be undefined. i think this is the correct behaviour, as we would have gotten a fatal error before when there are no constraints.

Comment (dantleech) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php]: would it make more sense to do this earlier if count($children) == 0 ?

File: lib/Doctrine/ODM/PHPCR/Query/Builder/ConstraintFactory.php [hunk 3]
```diff
@@ -127,7 +138,7 @@ public function fullTextSearch($field, $fullTextSearchExpression)
      * Same document constraint.
      *
      * <code>
-     * $qb->where()->same('/path/to/doc', 'sel_1');
+     * $qb->where()->same('/path/to/doc', 'sel_1')->end();
```
Comment (dantleech) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/ConstraintFactory.php]: ftr, the terminating `end()` is not necessary here

File: lib/Doctrine/ODM/PHPCR/Query/Builder/OperandDynamicFactory.php [hunk 4]
```diff
@@ -59,9 +59,11 @@ public function fullTextSearchScore($alias)
      * $qb->where()
      *   ->gt()
      *     ->length('alias_1.prop_1')
-     *     ->literal(50);
+     *     ->literal(50)
+     *   ->end()
+     * ->end();
      *
-     * $qb->orderBy()->asc()->fullTextSearchScore('sel_1');
+     * $qb->orderBy()->asc()->fullTextSearchScore('sel_1')->end();
```
Comment (dantleech) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/OperandDynamicFactory.php]: again, `end` is only necessary when chaining more stuff

**Comment by dantleech** [comment]
looks good, I would normally advise against adding the `end()` when it is not needed, as not having it simplifies things, but then not sure if that helps or hinders people (i.e. having to explain why it is not necessary there, but necessary when you have more stuff in the same line)

File: lib/Doctrine/ODM/PHPCR/Query/Builder/ConstraintFactory.php (hunk 3 above)
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/ConstraintFactory.php]: i know, but we have them in some points but not others. i thought having them everywhere makes it easier to copy-paste fragments. do we have a doc section explaining when end() is needed? and does absolutely everything have end() or do we need to be more explicit with that?

File: lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php [hunk 5]
```diff
@@ -513,13 +515,16 @@ protected function doWalkConstraintComposite(AbstractNode $node, $method)
...
         $children = $node->getChildren();
 
-        if (count($children) == 1) {
-            $op = $this->dispatch(current($children));
-            return $op;
+        if (0 === count($children)) {
+            throw new \InvalidArgumentException('Composite must have at least one constraint');
+        }
+        if (1 === count($children)) {
+            return $this->dispatch(current($children));
```
Comment (dbu) [inline lib/Doctrine/ODM/PHPCR/Query/Builder/BuilderConverterPhpcr.php]: good idea. also simplified the case of exactly one child a bit.

**Comment by dbu** [comment]
is it ok if we merge this as is, and at some point somebody writes an introduction explaining the `->end()` bits? we need that either way.

my reason to always specify end() here is that people will use this doc to copy-paste fragments of their queries.

**Comment by dantleech** [comment]
Yeah thats fine. We should [regenerate the generated documentation](https://github.com/doctrine/phpcr-odm/blob/master/lib/Doctrine/ODM/PHPCR/Tools/Console/Command/DumpQueryBuilderReferenceCommand.php) after this is merged.

**Comment by dbu** [comment]
good point. done in https://github.com/doctrine/phpcr-odm-documentation/commit/a5a56d8db7222fd002b0767f072541d5a01bf292

**Pull request merged**